Fixed a regression with the preserveJoinData option

EtienneBourque opened the pull request:
Hey!

This very simple pull request fixes a regression introduced by #42.

The original intent of the `_modifyEntity` method is to enter the `if` when `$object` is an instance of `BelongsToMany` and the `else` if not.

Now, when activating the `preserveJoinData` option, the `if` part is skipped correctly, but we enter the `else` part even if `$object` is an instance of `BelongsToMany`, which clears primary keys and causes duplicates of related entities in the database.

With this pull request, we check if `$object` is not an instance of `BelongsToMany` before clearing primary keys.

EtienneBourque committed:
Fixed a regression with the preserveJoinData option

EtienneBourque committed:
Changed else if for elseif

ADmad commented:
The fix makes sense. Could you please add a test case too to avoid regression in future.

EtienneBourque committed:
Improved test cases for preserveJoinData

EtienneBourque commented:
> The fix makes sense. Could you please add a test case too to avoid regression in future.

Good idea.

I changed the test a little because the test added in #42 used `duplicateEntity` which does not save in the database by itself.

I used `duplicate` instead like the other tests and now we can check for duplicates in the BelongsToMany association, in this case, tags.

I removed the first part of the test which was validating with `preserveJoinData` at `false` since it is already tested in the main test at lines 111 and 112.

Now, without the fix in this pull request, we fail at line 230 for having 4 tags instead of 2.
